src/librustc_codegen_llvm/consts.rs

```diff
@@ -209,7 +209,9 @@ impl CodegenCx<'ll, 'tcx> {
 
         debug!("get_static: sym={} instance={:?}", sym, instance);
 
-        let g = if let Some(id) = self.tcx.hir().as_local_hir_id(def_id) {
+        let g = if let Some(id) =
+            def_id.as_local().map(|def_id| self.tcx.hir().as_local_hir_id(def_id))
+        {
```

Comment:
Wouldn't it be cleaner to shadow `def_id` with the local version and then `let id = self.tcx.hir().as_local_hir_id(def_id);`?